at_onboarding_flutter/lib/screen/at_onboarding_activate_screen.dart

```diff
@@ -0,0 +1,212 @@
+        .loginWithAtsign(atsign ?? (widget.atSign ?? ''));
+    if (response.statusCode == 200) {
+      data = response.body;
+      data = jsonDecode(data);
+
+      final result = await AtOnboardingOTPScreen.push(
+        context: context,
+        atSign: atsign ?? (widget.atSign ?? ''),
+        hideReferences: false,
+      );
+
+      if (result != null) {
+        String? secret = result.secret?.split(':').last ?? '';
+        _processSharedSecret(atsign: result.atSign, secret: secret);
+      } else {
+        if (!mounted) return;
+        Navigator.pop(context, AtOnboardingResult.cancelled());
+      }
+    } else {
+      data = response.body;
+      data = jsonDecode(data);
+      String errorMessage = data['message'];
+      await showErrorDialog(errorMessage);
+    }
+  }
+
+  Future<void> showErrorDialog(String? errorMessage) async {
+    return AtOnboardingDialog.showError(
+      context: context,
+      message: errorMessage ?? '',
+      onCancel: () {
+        Navigator.of(context).pop();
+      },
+    );
+  }
+
+  void _showReferenceWebview() {
+    if (Platform.isAndroid || Platform.isIOS) {
+      AtOnboardingReferenceScreen.push(
+        context: context,
+        title: AtOnboardingStrings.faqTitle,
+        url: AtOnboardingStrings.faqUrl,
+      );
+    } else {
+      launchUrl(
+        Uri.parse(
+          AtOnboardingStrings.faqUrl,
+        ),
+      );
+    }
+  }
+
+  Future<dynamic> _processSharedSecret({
+    required String atsign,
+    required String secret,
+  }) async {
+    dynamic authResponse;
+    try {
+      atsign = atsign.startsWith('@') ? atsign : '@$atsign';
+
+      bool isExist = await _onboardingService.isExistingAtsign(atsign);
+      if (isExist) {
+        await _showAlertDialog(
+            AtOnboardingErrorToString().pairedAtsign(atsign));
+        return;
+      }
+      await Future.delayed(const Duration(seconds: 10));
+      authResponse = await _onboardingService.authenticate(atsign,
+          cramSecret: secret, status: OnboardingStatus.ACTIVATE);
+      if (authResponse == AtOnboardingResponseStatus.authSuccess) {
+        await AtOnboardingBackupScreen.push(context: context);
+        if (!mounted) return;
+        Navigator.pop(context, AtOnboardingResult.success(atsign: atsign));
+      } else if (authResponse == AtOnboardingResponseStatus.serverNotReached) {
+        await _showAlertDialog(
+          AtOnboardingStrings.atsignNotFound,
+        );
+      } else if (authResponse == AtOnboardingResponseStatus.authFailed) {
+        await _showAlertDialog(
+          AtOnboardingStrings.atsignNull,
+        );
+      } else {
+        await showErrorDialog('Response Time out');
+      }
+    } catch (e) {
+      if (e == AtOnboardingResponseStatus.authFailed) {
+        await _showAlertDialog(e, title: 'Auth Failed');
+      } else if (e == AtOnboardingResponseStatus.serverNotReached) {
+        await _processSharedSecret(atsign: atsign, secret: secret);
```

Comment:
@sonle-geekyants As we are calling the same function from inside of this function, can we run into a scenario where this function keeps on calling itself.